gazebo-web/gazebosim-web-frontend
# Default bad URLs to not-found page

Fix #16. Yay!
Depends on https://github.com/gazebo-web/gazebosim-web-frontend/pull/25 , which fixes the second part (versioned URLs under `docs/`) of #16. Will rebase after that is merged.

This PR fixes the first part of #16. Namely, these two cases:
- Bad URL in the general case, e.g. http://localhost:3000/bad_url . The fix is in `app-routing.module.ts`. Previously, `**` URLs default to the home page. Now they default to the `not-found` page created in #25.
- Bad URL that is non-versioned under docs/, e.g. http://localhost:3000/docs/bad_url . The fix is in `doc/doc.component.ts`. Previously, if a version is not found, it just defaults to the latest version. Now it defaults to go to `not-found`.

## To test

I used the `docker_dev` branch in both gazebosim-web-frontend and gazebosim-web-backend repos to test it (merge to that branch locally):
```
cd path/to/gazeobsim-web-frontend
docker build . -t gazebosim_web_frontend
cd path/to/gazebosim-web-backend
docker build . -t gazebosim_web_backend
cd path/to/gazeobsim-web-frontend
docker-compose up
```
Type in the 2 URLs in bullet list above. They should lead to the `not-found` page.

@azeey to review.
